EIPS/eip-747.md

```diff
@@ -35,26 +35,14 @@ The `wallet_watchAsset` method takes a single parameter, a `WatchAssetParameters
 
 ```typescript
 interface WatchAssetParameters {
-  type: string; // The asset's interface, e.g. 'ERC20'
-  options: {
-    address: string; // The hexadecimal Ethereum address of the token contract
-    chainId?: number; // The chain ID of the asset. If empty, defaults to the current chain ID.
-    image?: string; // A string url of the token logo
-  };
+  type: string; // The asset's interface, e.g. 'ERC1046'
+  options: any;
```

Comment:
Don't use `any`, it tells the compiler "disable type checking".
```suggestion
  options: unknown
```
It is also possible (likely) that you really mean empty object:
```suggestion
  options: {}
```

Reply:
I want this to be arbitrarily extensible, so `any` is in fact correct here.

Reply:
You want `unknown` or `{}` then.  `any` is **never** the right type.  `any` tells the type checker "don't do any type checking" and it exists only for backward compatibility with JavaScript and shouldn't be used by anyone writing new code (or specifications).

If you want `options` to be allowed to be a `string` or a `number` or a `boolean` then you should use `unknown`.  If you want `options` to always be an object with string keys, then `Record<string, unknown>`.  If you want `options` to be allowed to have number or symbol keys, then `{}` is correct.

Note that setting the type to `{}` doesn't mean you can *only* pass an empty object.  TypeScript is duck typed, which means that `{ a: 'a' }` can be assigned to a variable with a type of `{}` because extra properties are allowed.